torch/profiler/_pattern_matcher.py

```diff
@@ -50,13 +51,28 @@ def eventTreeTraversal(self):
     def summary(self, events: List[_ProfilerEvent]):
         default_summary = f"{self.name}: {len(events)} events matched."
         if self.should_benchmark:
-            summary = self.benchmark_summary(events)
             # If benchmark summary is not empty, use it.
-            return summary if summary else default_summary
+            return self.benchmark_summary(events) if hasattr(  # type: ignore[attr-defined]
+                self, 'benchmark') else default_summary
         return default_summary
 
     def benchmark_summary(self, events: List[_ProfilerEvent]):
-        return ""
+        def format_time(time_ns: int):
+            unit_lst = ["ns", "us", "ms", "s"]
+            for unit in unit_lst:
+                if time_ns < 1000:
+                    return f"{time_ns:.2f} {unit}"
+                time_ns //= 1000
+
+        assert hasattr(self, 'benchmark'), 'Please implement benchmark()'
```

Comment:
Add a base `benchmark` method that raises `NotImplementedError`?

Reply:
I need to check if we have implemented benchmark in summary(), I haven't find a better way to solve this